File: Java/SearchingAndSorting/Problem28/Arithmetic Number.java
package SearchingAndSorting.Problem28;

class Solution {
    public static void main(String[] args) {
        System.out.println(arithmeticNumber(1, 3, 2));
        System.out.println(arithmeticNumber(1, 2, 3));
        System.out.println(arithmeticNumber(1, 2, 4));
    }

    private static boolean arithmeticNumber(Integer a, Integer b, Integer c) {
        double lhs = (double) (b - a) /c;
        return lhs == (int) lhs;
    }
}

File: Python/Practice Set 3/Searching & Sorting/Arithmetic Number.py
# Problem link - https://www.geeksforgeeks.org/problems/arithmetic-number2815/1


class Solution:
    @staticmethod
    def arithmetic_number(a, b, c):
        """
            Time and space complexities both are O(1).
        """
        lhs = (b - a)/c
        return lhs == int(lhs)


print(Solution.arithmetic_number(1, 3, 2))
print(Solution.arithmetic_number(1, 2, 3))
print(Solution.arithmetic_number(1, 2, 4))